pkg/db/migration/convert/convert000031/convert.go

```diff
@@ -0,0 +1,104 @@
+package convert000031
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+
+	"github.com/instill-ai/pipeline-backend/pkg/datamodel"
+)
+
+const batchSize = 100
+
+type ConvertToWeb struct {
+	DB     *gorm.DB
+	Logger *zap.Logger
+}
+
+func (c *ConvertToWeb) Migrate() error {
+	log.Println("=== Start migration ===")
+	if err := c.migratePipeline(); err != nil {
+		return err
+	}
+
+	return c.migratePipelineRelease()
+}
+
+func (c *ConvertToWeb) migratePipeline() error {
+	pipelines := make([]*datamodel.Pipeline, 0, batchSize)
+	return c.DB.Select("uid", "recipe_yaml", "recipe").FindInBatches(&pipelines, batchSize, func(tx *gorm.DB, _ int) error {
+		for _, p := range pipelines {
+			l := c.Logger.With(zap.String("pipelineUID", p.UID.String()))
+
+			recipeYAML := p.RecipeYAML
+
+			updatedRecipeYAML, isUpdate := c.updateRecipeYAML(recipeYAML)
+
+			log.Println("Updated Recipe YAML: \n", updatedRecipeYAML)
+
+			log.Println("Is Update: ", isUpdate)
+
+			if isUpdate {
+				result := tx.Model(p).Where("uid = ?", p.UID).Update("recipe_yaml", updatedRecipeYAML)
+				log.Println("=== Updating Pipeline ===")
+
+				if result.Error != nil {
+					l.Error("Failed to update pipeline release.")
+					return fmt.Errorf("updating pipeline recipe: %w", result.Error)
+				}
+			}
+		}
+
+		return nil
+	}).Error
+}
+
+func (c *ConvertToWeb) migratePipelineRelease() error {
+	pipelineReleases := make([]*datamodel.PipelineRelease, 0, batchSize)
+	return c.DB.Select("uid", "recipe_yaml", "recipe").FindInBatches(&pipelineReleases, batchSize, func(tx *gorm.DB, _ int) error {
+		for _, p := range pipelineReleases {
+			l := c.Logger.With(zap.String("pipelineUID", p.UID.String()))
+
+			recipeYAML := p.RecipeYAML
+
+			updatedRecipeYAML, isUpdate := c.updateRecipeYAML(recipeYAML)
+
+			if isUpdate {
+				result := tx.Model(p).Where("uid = ?", p.UID).Update("recipe_yaml", updatedRecipeYAML)
+				log.Println("=== Updating PipelineRelease ===")
+				if result.Error != nil {
+					l.Error("Failed to update pipeline release.")
+					return fmt.Errorf("updating pipeline recipe: %w", result.Error)
+				}
+			}
+		}
+
+		return nil
+	}).Error
+}
+
+func (c *ConvertToWeb) updateRecipeYAML(recipeYAML string) (updatedRecipeYAML string, isUpdate bool) {
+
+	if !strings.Contains(recipeYAML, "TASK_CRAWL_WEBSITE") && !strings.Contains(recipeYAML, "TASK_SCRAPE_WEBPAGE") {
+		return "", false
+	}
+
+	if strings.Contains(recipeYAML, "TASK_CRAWL_WEBSITE") {
+		// Insert the description of the recipe needs to be updated in the top of the YAML file
+		updatedRecipeYAML = fmt.Sprintf("# Recipe updated to replace TASK_CRAWL_WEBSITE with TASK_CRAWL_SITE and deprecated some fields. \n# Please check the latest document to update your recipe.\n%s", recipeYAML)
+
+		updatedRecipeYAML = strings.ReplaceAll(updatedRecipeYAML, "TASK_CRAWL_WEBSITE", "TASK_CRAWL_SITE")
+		updatedRecipeYAML = strings.ReplaceAll(updatedRecipeYAML, "target-url", "url")
```

Comment:
How are you sure you're only replacing the `url` field in this component and not other parts of the recipe (e.g. an input variable)?

Reply:
I missed it. I removed it.
It will need users to update those deprecated recipes

Reply:
@chuang8511 
I believe we need to migrate this. Ideally, you can use a YAML parser to accurately locate the web component and update both the task and target URL.

Reply:
Yes, that’s an issue. Since migration 31 already do this, let’s go ahead and use the JSON approach for the update this time. However, @chuang8511 , please ensure that you only update the pipelines containing the web component.

Reply:
@donch1989 
> Downside: we'll lose the comments in the updated recipes

Yes, that is the reason why I changed to the current way.
It seems there is no easy way to keep the comments.
So, is it acceptable?
Or, do you have other reference that we can follow?

I will keep searching if there is a simple way to achieve it.